synapse/events/builder.py

```diff
@@ -128,6 +128,7 @@ async def build(
                 state_filter=StateFilter.from_types(
                     auth_types_for_event(self.room_version, self)
                 ),
+                await_full_state=False,
```

Comment:
`EventBuilder.build` seems to be only called in `MessageHandler` so it should be safe to default to `await_full_state=False` there.
Or do we want to add an explicit parameter instead to avoid future potential misuse of `build` outside of the message handler ?

Reply:
If we calculate `await_full_state` based on the `state_filter` (see other comment), we should be able to avoid misuse.

Though that will block if a lazy joined admin tries to kick someone from the room, so maybe we do need `await_full_state=False` after all...

Reply:
I think this is fine as is.